interface/patient_file/pos_checkout_ippf.php

```diff
@@ -0,0 +1,2798 @@
+            <?php echo xlt('Tentative Invoice Ref No'); ?>
+        </td>
+        <td class='text' align='right' colspan='<?php echo $form_num_amount_columns; ?>'>
+            <?php echo $irnumber; ?>
+        </td>
+    </tr>
+        <?php
+    } else if (!empty($GLOBALS['gbl_mask_invoice_number'])) {
+    // Otherwise if there is an invoice reference number mask, ask for the refno.
+        ?>
+    <tr>
+        <td class='bold' colspan='<?php echo ($form_num_type_columns + $form_num_method_columns + $form_num_ref_columns) +
+            (empty($GLOBALS['gbl_charge_categories']) ? 0 : 1); ?>' align='right'>
+            <?php echo xlt('Invoice Reference Number'); ?>
+        </td>
+        <td class='text' align='right' colspan='<?php echo $form_num_amount_columns; ?>'>
+            <input type='text' name='form_irnumber' size='10' value=''
+                onkeyup='maskkeyup(this,"<?php echo addslashes($GLOBALS['gbl_mask_invoice_number']); ?>")'
+                onblur='maskblur(this,"<?php echo addslashes($GLOBALS['gbl_mask_invoice_number']); ?>")'
+        />
+        </td>
+    </tr>
+        <?php
+    }
+}
+?>
+
+    <tr>
+        <td class='text' colspan='<?php echo 5 + $num_optional_columns; ?>' align='center'>
+            &nbsp;<br>
+            <input type='submit' name='form_save' value='<?php echo xlt('Save'); ?>'
+<?php if ($rapid_data_entry) { ?>
+                style='background-color:#cc0000';color:#ffffff'
+<?php } ?>
+            /> &nbsp;
+<?php if (empty($_GET['framed'])) { ?>
+            <input type='button' value='Cancel' onclick='window.close()' />
+<?php } ?>
+            <input type='hidden' name='form_provider'  value='<?php echo $inv_provider  ?>' />
+            <input type='hidden' name='form_payer'     value='<?php echo $inv_payer     ?>' />
+            <input type='hidden' name='form_encounter' value='<?php echo $encounter_id ?>' />
+            <input type='hidden' name='form_checksum'  value='<?php echo $current_checksum; ?>' />
+        </td>
+    </tr>
+
+</table>
+
+</center>
+
+</form>
+
+<script>
+
+    // Add a line for entering a payment.
+    // Declared down here because $form_num_*_columns must be defined.
+    var paylino = 0;
+    function addPayLine() {
+        var table = document.getElementById('paytable');
+        for (var i = 0; i < table.rows.length; ++i) {
+            if (table.rows[i].id == 'totalpay') {
+                var row = table.insertRow(i);
+                var cell;
+<?php
+foreach ($aCellHTML as $ix => $html) {
+    echo "    var html = \"$html\";\n";
+    echo "    cell = row.insertCell(row.cells.length);\n";
+    if ($ix == 0) {
+        echo "    cell.colSpan = $form_num_type_columns;\n";
+    }
+    if ($ix == 1) {
+        echo "    cell.colSpan = $form_num_method_columns;\n";
+    }
+    if ($ix == 2) {
+        echo "    cell.colSpan = $form_num_ref_columns;\n";
+    }
+    if ($ix == 3) {
+        echo "    cell.colSpan = $form_num_amount_columns;\n";
```

Comment:
is these are going out to javascript context, then would wrap the variables with js_escape()

Reply:
These are hard coded to integer values.